pkg/build/builder/docker.go

```diff
@@ -70,40 +66,27 @@ func (d *DockerBuilder) Build() error {
 		return err
 	}
 	glog.V(4).Infof("Starting Docker build from %s/%s BuildConfig ...", d.build.Namespace, d.build.Name)
-	var push bool
-
-	// if there is no output target, set one up so the docker build logic
-	// will still work, but we won't push it at the end.
-	if d.build.Spec.Output.To == nil || len(d.build.Spec.Output.To.Name) == 0 {
-		d.build.Spec.Output.To = &kapi.ObjectReference{
-			Kind: "DockerImage",
-			Name: noOutputDefaultTag,
```

Comment:
@bparees, if d.build.Status.OutputDockerImageReference is empty, the build
should succeed  (or fail for some other reason), but there should be no
push (and this shall be exercised by 'oc new-build <source> --no-output' --
new feature Clayton wants #5296)
On Oct 22, 2015 5:35 PM, "Ben Parees" <notifications@github.com> wrote:

In pkg/build/builder/docker.go
<https://github.com/openshift/origin/pull/5071#discussion_r42764226>:

> @@ -70,40 +66,27 @@ func (d *DockerBuilder) Build() error {
>  		return err
>  	}
>  	glog.V(4).Infof("Starting Docker build from %s/%s BuildConfig ...", d.build.Namespace, d.build.Name)
> -	var push bool
> -
> -	// if there is no output target, set one up so the docker build logic
> -	// will still work, but we won't push it at the end.
> -	if d.build.Spec.Output.To == nil || len(d.build.Spec.Output.To.Name
) == 0 {
> -		d.build.Spec.Output.To = &kapi.ObjectReference{
> -			Kind: "DockerImage",
> -			Name: noOutputDefaultTag,

isn't this still going to be a problem for the build, if no
d.build.Status.OutputDockerImageReference is set, i think the build is
going to fail?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5071/files#r42764226>.